registry/pkg-metadata-filter/src/index.ts

```diff
@@ -1,17 +1,34 @@
-import { globalWarn } from '@pnpm/logger'
+import { globalInfo, globalWarn } from '@pnpm/logger'
 import { type PackageMetadataWithTime } from '@pnpm/registry.types'
 import semver from 'semver'
 
+// Track packages that have already been logged to avoid duplicates
+const loggedPackages = new Set<string>()
```

Comment:
`pickPackageFromMetaUsingTime` may call this function twice for the same package.


https://github.com/pnpm/pnpm/blob/102d5a01ddabda1184b88119adccfbe956d30579/resolving/npm-resolver/src/pickPackage.ts#L84-L86